conn_http.go

```diff
@@ -265,7 +267,7 @@ func (h *httpConnect) isBad() bool {
 }
 
 func (h *httpConnect) readTimeZone(ctx context.Context) (*time.Location, error) {
-	rows, err := h.query(ctx, func(*connect, error) {}, "SELECT timezone()")
+	rows, err := h.query(Context(ctx, ignoreExternalTables()), func(*connect, error) {}, "SELECT timezone()")
```

Comment:
Since it's an internal logic for HTTP proto, we should not leak it into Options struct.

Instead, can we reset `external` in context? Or can we pass implicit bool to `query` func?

Reply:
looks like `query` is defined in `stdConnect` interface with a generic signature, modify `context` from caller side seems a more reasonable way to optimize internal calls for timezone and ping.